action.yml

```diff
@@ -17,6 +17,10 @@ runs:
       shell: sh
       if: runner.os == 'Linux'
       run: |
+        for f in $(chmod -c -R +r . | awk '{print substr($3, 2, length($3)-2)}')
```

Comment:
If a filename has a blank in it, this'll pick up only the part of the name before the blank; a slightly more robust way could be to use `'` as the field separator and loop over entire lines instead of splitting on words:

```sh
chmod -c -R +r . | awk -F\' '{ print $2 }' \
    | while IFS= read -r f; do
        echo "::warning::Added read permission to $f"
    done
```

but that breaks if the filename contains quotes. With GNU grep, a solution that should be able to handle all characters in filenames could look like

```sh
chmod -c -R +r . | grep -Po '^mode of (["'\''])\K.*(?=\1 changed from \d+)'
```

handling blanks, single and double quotes, and newlines in filenames, printing like

```sh
./subdir/file
./file'$'\n''newline
./file"'\''quotes
./file'quote
./has space
```

but that would require to rewrite the loop to

```sh
chmod -c -R +r . \
    | grep -Po '^mode of (["'\''])\K.*(?=\1 changed from \d+)' \
    | while IFS= read -r f; do
        echo "::warning::Added read permission to $f"
    done
```

This still works in `sh`, but I understand the action tries to support more limited environments like busybox, where grep won't have the `-P` option.

Probably not super important because it only affects the warning message, not the actual `chmod` operation, but I thought I'd point it out.

Reply:
Kind of like the suggestion of not parsing at all and just wrapping the output in a warning 👍